Targets/Mashow/Sources/Util/Login/AuthorizationManager+Apple.swift

```diff
@@ -6,11 +6,70 @@
 //  Copyright © 2024 com.alcoholers. All rights reserved.
 //
 
-import Foundation
+import UIKit
+import AuthenticationServices
 
 extension AuthorizationManager {
-    // FIXME: 잘 고쳐줘 다연아
-    func signInWithApple() async throws -> (idToken: String, accessToken: String) {
-        fatalError()
+    func signInWithApple(_ view: UIViewController) async throws -> String {
+        let appleAuthManager = AppleAuthManager()
+        return try await appleAuthManager.didTapSignInWithAppleButton(from: view)
+    }
+}
+
+final class AppleAuthManager: NSObject {
+    weak var viewController: UIViewController?
+    private var completionHandler: ((Result<String, Error>) -> Void)? = nil
+    
+    func setAppleAuthPresentationAnchorView(_ view: UIViewController) {
+        self.viewController = view
+    }
+    
+    func didTapSignInWithAppleButton(from view: UIViewController) async throws -> String {
+        setAppleAuthPresentationAnchorView(view)
+        return try await withCheckedThrowingContinuation { continuation in
+            startAuthFlow { result in
+                switch result {
+                case .success(let identityToken):
+                    continuation.resume(returning: identityToken)
+                    return
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                    return
+                }
+            }
+        }
+    }
+    
+    private func startAuthFlow(completion: @escaping (Result<String, Error>) -> Void) {
+        completionHandler = completion
+        let provider = ASAuthorizationAppleIDProvider()
+        let request = provider.createRequest()
+        request.requestedScopes = [.email]
+        
+        let controller = ASAuthorizationController(authorizationRequests: [request])
+        controller.delegate = self
+        controller.presentationContextProvider = self
+        controller.performRequests()
+    }
+}
+
+extension AppleAuthManager: ASAuthorizationControllerDelegate {
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        if let credential = authorization.credential as? ASAuthorizationAppleIDCredential,
+           let idToken = credential.identityToken,
+           let idTokenString = String(data: idToken, encoding: .utf8) {
+            completionHandler?(.success(idTokenString))
+        }
+    }
+    
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: any Error) {
+        print(error.localizedDescription)
+        completionHandler?(.failure(error))
+    }
+}
+
+extension AppleAuthManager: ASAuthorizationControllerPresentationContextProviding {
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        return viewController!.view.window!
```

Comment:
```suggestion
        guard
            let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
                let window = windowScene.windows.first
        else {           
            print("No window available for presentation.")
            return UIWindow()
        }
```
지금처럼 ViewController를 명시적으로 주는 방법도 좋다고 생각합니당. 보자마자 맥락을 파악할 수 있는게 좋은 코드라고 생각해서ㅋㅋ
근데도 영 찝찝하면 이렇게 지금 보이는 window를 바로 가져오는 방법도 있긴 한데너가 좋은 걸로 쓰면 될 것 같음!

(지금 생각해보니까 Logger도 하나 만들어야 할 듯. 에러 프린트를 매번 `print`로 하기엔 너무 번거롭..)

Reply:
이놈으로 가겠다 .